src/scripts/main.js

```diff
@@ -2,4 +2,27 @@
 
 document.addEventListener('click', (e) => {
   // write code here
+  const spider = document.querySelector('.spider');
+  const wall = e.target.closest('.wall');
+
+  if (!wall) {
+    return;
+  }
+
+  const wallRect = wall.getBoundingClientRect();
+  const borderSize = parseInt(getComputedStyle(wall).borderWidth);
+  const spiderRect = spider.getBoundingClientRect();
+
+  let targetX = e.clientX - wallRect.left - spiderRect.width / 2 - borderSize;
+  let targetY = e.clientY - wallRect.top - spiderRect.height / 2 - borderSize;
+
+  targetX = Math.max(0, Math.min(targetX, wall.clientWidth - spiderRect.width));
```

Comment:
The logic for ensuring that the spider doesn't go outside the wall can be a bit hard to follow. Consider extracting this into a separate function, like `keepSpiderInBounds`, to improve readability.